internal/services/network/express_route_circuit_peering_resource.go

```diff
@@ -58,12 +58,29 @@ func resourceExpressRouteCircuitPeering() *pluginsdk.Resource {
 
 			"primary_peer_address_prefix": {
 				Type:     pluginsdk.TypeString,
-				Required: true,
+				Optional: true,
+				RequiredWith: []string{
+					"secondary_peer_address_prefix",
+					"ipv4_enabled",
+				},
 			},
 
 			"secondary_peer_address_prefix": {
 				Type:     pluginsdk.TypeString,
-				Required: true,
+				Optional: true,
+				RequiredWith: []string{
+					"primary_peer_address_prefix",
+					"ipv4_enabled",
+				},
+			},
+
+			"ipv4_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				RequiredWith: []string{
```

Comment:
this makes this a breaking change?

Reply:
yes, this is breaking change. If user configured the `ipv4` block, service will set this flag to true by default, should we change this to `compute=true`,  this will not brought any changes and compatible with the former state files.

Reply:
I think we have to otherwise it'll be a breaking change?

Reply:
change `ipv4_enabled, ipv6_enabled` to `optional` and defaults to `true`.

Reply:
```log
=== RUN   TestAccExpressRouteCircuitConnection
=== RUN   TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringIpv6WithRouteFilter
=== RUN   TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeering
=== RUN   TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringCustomerRouting
--- PASS: TestAccExpressRouteCircuit (3374.38s)
    --- PASS: TestAccExpressRouteCircuit/PrivatePeering (1169.73s)
        --- PASS: TestAccExpressRouteCircuit/PrivatePeering/azurePrivatePeering (367.62s)
        --- PASS: TestAccExpressRouteCircuit/PrivatePeering/azurePrivatePeeringWithUpdate (442.81s)
        --- PASS: TestAccExpressRouteCircuit/PrivatePeering/requiresImport (359.30s)
    --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering (2204.65s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringWithRouteFilter (372.12s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringIpv6 (380.37s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringIpv6CustomerRouting (344.60s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringIpv6WithRouteFilter (465.15s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeering (372.53s)
        --- PASS: TestAccExpressRouteCircuit/MicrosoftPeering/microsoftPeeringCustomerRouting (269.88s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       3397.623s
```